java/code/src/com/redhat/rhn/taskomatic/task/HubReportDbUpdateWorker.java

```diff
@@ -0,0 +1,154 @@
+            WriteMode delete = generateDelete(localSession, tableName);
+            delete.executeUpdate(Map.of("mgm_id", mgmId));
+
+            // Extract the first batch
+            @SuppressWarnings("unchecked")
+            DataResult<Map<String, Object>> firstBatch = query.execute(Map.of("offset", 0, "limit", BATCH_SIZE));
+            firstBatch.forEach(e -> e.remove("mgm_id"));
+
+            if (!firstBatch.isEmpty()) {
+                // Generate the insert using the column name retrieved from the select
+                WriteMode insert = generateInsert(localSession, tableName, mgmId, firstBatch.get(0).keySet());
+                insert.executeUpdates(firstBatch);
+                LogMF.debug(log, "Extracted {} rows for table {}", firstBatch.size(), tableName);
+
+                // Iterate further if we can have additional rows
+                if (firstBatch.size() == BATCH_SIZE) {
+                    ReportDBHelper.<Map<String, Object>>batchStream(query, BATCH_SIZE, BATCH_SIZE)
+                            .forEach(batch -> {
+                                batch.forEach(e -> e.remove("mgm_id"));
+                                insert.executeUpdates(batch);
+                                LogMF.debug(log, "Extracted {} rows more for table {}", firstBatch.size(), tableName);
+                            });
+                }
+            }
+            else {
+                LogMF.debug(log, "No data extracted for table {}", tableName);
+            }
+        });
+    }
+
+    @Override
+    public void run() {
+        try {
+            parentQueue.workerStarting();
+            ConnectionManager localRcm = ConnectionManagerFactory.localReportingConnectionManager();
+            ReportDbHibernateFactory localRh = new ReportDbHibernateFactory(localRcm);
+            Credentials credentials = mgrServerInfo.getReportDbCredentials();
+            ConnectionManager remoteDBCM = ConnectionManagerFactory.reportingConnectionManager(
+                    credentials.getUsername(), credentials.getPassword(),
+                    "jdbc:postgresql://" + mgrServerInfo.getReportDbHost() + ":" +
+                            mgrServerInfo.getReportDbPort() + "/" + mgrServerInfo.getReportDbName()
+                    );
+            ReportDbHibernateFactory remoteDB = new ReportDbHibernateFactory(remoteDBCM);
+            try {
+
+                TABLES.forEach(table -> {
+                    updateRemoteData(remoteDB.getSession(), localRh.getSession(), table, mgrServerInfo.getId());
+                });
+                mgrServerInfo.setReportDbLastSynced(new Date());
+                ServerFactory.save(mgrServerInfo.getServer());
+                localRcm.commitTransaction();
+                log.info("Reporting db updated for server " + mgrServerInfo.getServer().getId() + " successfully.");
+            }
+            catch (RuntimeException ex) {
+                log.warn("Unable to update reporting db", ex);
```

Comment:
I see that we are not re-throwing the exception in case of error. Is the job going to be marked as failed in this case or will it look always succeed?

Reply:
It should at least that's what happened when i got exceptions during development. The way the try catch is structured here was copied from the erratacacheworker so it should hopefully be right for those type of tasks.